### PR TITLE
fix(web.domain): change management renewable vs restorable

### PR DESCRIPTION
**Pull request by stif59100**
ref:MANAGER-9563

Signed-off-by: stif59100 <steeve.vanderstocken@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `release/sprint-2248-2250` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-9563
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~~Breaking change is mentioned in relevant commits~~~

## Description

I change management on domain for renewable and restorable.

## Related

:flags: Translation
- [x] TRANS-48183

### CODE REVIEW

File: packages/manager/apps/web/client/app/components/service-status/service-status-action.html [hunk 1]
```diff
@@ -38,13 +38,21 @@
     </dl>
     <oui-action-menu data-compact data-placement="end">
         <oui-action-menu-item
-            data-ng-if="$ctrl.isExpired() || !$ctrl.isInAutoRenew()"
+            data-ng-if="($ctrl.isExpired() && !$ctrl.isRestorable()) || (!$ctrl.isInAutoRenew() && !$ctrl.isRestorable())"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/components/service-status/service-status-action.html:41]:
```suggestion
            data-ng-if="!$ctrl.isRestorable() && (!$ctrl.isRestorable() || !$ctrl.isInAutoRenew())"
```

File: packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html [hunk 2]
```diff
@@ -30,7 +30,9 @@
                 </span>
             </div>
         </div>
-        <div data-ng-if="$ctrl.isExpired() || !$ctrl.isInAutoRenew()">
+        <div
+            data-ng-if="($ctrl.isExpired() && !$ctrl.isRestorable()) || (!$ctrl.isInAutoRenew && !$ctrl.isRestorable())"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html:34]: same as here:
https://github.com/ovh/manager/pull/8517/files#r1041963952

File: packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html [hunk 3]
```diff
@@ -56,5 +58,33 @@
                 ></span>
             </a>
         </div>
+        <div
+            data-ng-if="($ctrl.isExpired() && $ctrl.isRestorable()) || (!$ctrl.isInAutoRenew && $ctrl.isRestorable())"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html:62]:
```suggestion
            data-ng-if="$ctrl.isRestorable() && ($ctrl.isExpired() || !$ctrl.isInAutoRenew())"
```

Comment (mohammed-zahaf) [reply packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html:62]: i saw `$ctrl.isInAutoRenew` but i think it's `$ctrl.isInAutoRenew()` ?

File: packages/manager/apps/web/client/app/components/service-status/service-status-action.html [hunk 4]
```diff
@@ -38,13 +38,21 @@
...
             data-aria-label="{{ ::'status_action_wizard_renew' | translate }}"
             data-href="{{ $ctrl.orderUrl }}"
             data-external
         >
             <span data-translate="status_action_wizard_renew"></span>
         </oui-action-menu-item>
+        <oui-action-menu-item
+            data-ng-if="($ctrl.isExpired() && $ctrl.isRestorable()) || (!$ctrl.isInAutoRenew() && $ctrl.isRestorable())"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/components/service-status/service-status-action.html:49]:
```suggestion
            data-ng-if="$ctrl.isRestorable() && ($ctrl.isExpired() || !$ctrl.isInAutoRenew())"
```

File: packages/manager/apps/web/client/app/domain/dashboard/domain.html [hunk 5]
```diff
@@ -146,20 +151,32 @@
...
         ></a>
     </div>
+    <div
+        class="alert alert-warning"
+        role="alert"
+        data-ng-if="ctrlDomain.domain.isExpired && ctrlDomain.domain.isRestorable"
+    >
+        <span
+            data-translate="common_service_restorable_1"
+            data-translate-values="{ t0: '<strong>' + ctrlDomain.domain.displayName + '</strong>' }"
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.html:174]: Same as https://github.com/ovh/manager/pull/8517/files#r1041975066

File: packages/manager/apps/web/client/app/domain/dashboard/domain.constant.js [hunk 6]
```diff
@@ -2,6 +2,32 @@ export const ALERTS = {
   tabs: 'domain_alert_tabs',
 };
 
+export const RENEW_URL = {
+  default: '/cgi-bin/order/renew.cgi?domainChooser=',
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.constant.js:6]:
```suggestion
  DEFAULT: '/cgi-bin/order/renew.cgi?domainChooser=',
```

**Review by anooparveti — COMMENTED** [review]
Could you please mention the ticket in the PR description?
Thanks!

File: packages/manager/apps/web/client/app/domain/dashboard/domain.controller.js [hunk 7]
```diff
@@ -137,11 +141,21 @@ angular.module('App').controller(
       );
     }
 
+    getRenewUrl() {
+      this.user = this.coreConfig.getUser();
+      return `${RENEW_URL[this.user.ovhSubsidiary] || RENEW_URL.default}${
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.controller.js:146]:
```suggestion
      return `${RENEW_URL[this.user.ovhSubsidiary] || RENEW_URL.DEFAULT}${
```

File: packages/manager/apps/web/client/app/domain/dashboard/domain.service.js [hunk 8]
```diff
@@ -145,6 +145,12 @@ angular.module('services').service(
       });
     }
 
+    getDomainCheckRegistry(domain) {
+      this.$http.get(`/registry/${domain}/domaincheck`, {
+        rootPath: 'apiv6',
+      });
+    }
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.service.js:152]:
```suggestion
    getDomainCheckRegistry(domain) {
      return this.$http.get(`/registry/${domain}/domaincheck`).then(({data}) => data);
    }
```

File: packages/manager/apps/web/client/app/domain/dashboard/domain.html [hunk 9]
```diff
@@ -146,20 +151,32 @@
...
+        ></span>
+        <a href="{{::ctrlDomain.getRenewUrl()}}" target="_blank"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.html:176]:
```suggestion
        <a href="{{::ctrlDomain.getRenewUrl()}}" target="_blank" rel="noopener"
```

File: packages/manager/modules/pci/src/projects/project/gateways/gateways.controller.js [hunk 10]
```diff
@@ -12,6 +12,7 @@ export default class PublicGatewaysController {
   $onInit() {
     this.loadMessages();
     this.formatIps();
+    console.log(this.gateways);
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/gateways/gateways.controller.js:15]: **niptick:** Could you please remove this?

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html [hunk 11]
```diff
@@ -30,7 +30,10 @@
                 </span>
             </div>
         </div>
-        <div data-ng-if="$ctrl.isExpired() || !$ctrl.isInAutoRenew()">
+
+        <div
+            data-ng-if="($ctrl.isExpired() || !$ctrl.isInAutoRenew()) && ($ctrl.serviceInfos.domain !== $ctrl.serviceName)"
```
Comment (ghyenne) [inline packages/manager/apps/web/client/app/components/expiration/service-expiration-date.component.html:35]:
```suggestion
            data-ng-if="($ctrl.isExpired() || !$ctrl.isInAutoRenew()) && $ctrl.serviceInfos.domain !== $ctrl.serviceName"
```
I think that the brackets are not necessary

File: packages/manager/apps/web/client/app/domain/dashboard/domain.constant.js [hunk 12]
```diff
@@ -2,6 +2,32 @@ export const ALERTS = {
   tabs: 'domain_alert_tabs',
 };
 
+export const RENEW_URL = {
+  DEFAULT: '/cgi-bin/order/renew.cgi?domainChooser=',
```
Comment (ghyenne) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.constant.js:6]: This url is not valid.
Also I think we can extract the suffix from the url

```suggestion
  const RENEW_URL_SUFFIX = 'cgi-bin/order/renew.cgi?domainChooser=';
  export const RENEW_URL = {
  AU: `https://ca.ovh.com/au/${RENEW_URL_SUFFIX}`,
  CA: `https://ca.ovh.com/fr/${RENEW_URL_SUFFIX}`,
  CZ: `https://www.ovh.cz/${RENEW_URL_SUFFIX}`,
```

File: packages/manager/apps/web/client/app/domain/zone/domain-zone-dns.html [hunk 13]
```diff
@@ -7,13 +7,13 @@
...
             <div class="col-md-9">
                 <div data-ovh-alert="{{alerts.main}}"></div>
-
                 <div
                     class="alert alert-info"
                     role="alert"
                     data-translate="domain_tab_ZONE_no_zone"
                     data-ng-if="ctrlDomainTabZoneDns.displayActivateZone"
                 ></div>
+
```
Comment (ghyenne) [inline packages/manager/apps/web/client/app/domain/zone/domain-zone-dns.html:16]: No need to commit a file for the removal/addition of a blank line.

**Review by ghyenne — CHANGES_REQUESTED** [review]

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/apps/web/client/app/domain/dashboard/domain.service.js [hunk 14]
```diff
@@ -135,6 +135,12 @@ angular.module('services').service(
       });
     }
 
+    getDomainsService(serviceName) {
+      return this.OvhHttp.get(`/domain/${serviceName}`, {
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/domain/dashboard/domain.service.js:139]: you could use `$http.get` instead.
```suggestion
      return this.$http.get(`/domain/${serviceName}`).then(({data}) => data)
```

Comment (stif59100) [reply packages/manager/apps/web/client/app/domain/dashboard/domain.service.js:139]: we use in this file OvhHttp

getTasks(serviceName, params = {}) {
      return this.OvhHttp.get(`/domain/${serviceName}/task`, {
      return this.OvhHttp.get(`/domain/${serviceName}/task`, {
        rootPath: 'apiv6',
        rootPath: 'apiv6',
        params,
        params,
      });
      });
    }
    }
    /**
    /**
     * Get tasks by status
     * Get tasks by status
     * @param {string} serviceName
     * @param {string} serviceName
     * @param {string} fn
     * @param {string} fn
     * @param {Array} status
     * @param {Array} status
     */
     */
    getTasksByStatus(serviceName, fn, status = []) {
    getTasksByStatus(serviceName, fn, status = []) {
      const promisesTasks = map(status, (st) =>
      const promisesTasks = map(status, (st) =>
        this.getTasks(serviceName, { status: st, function: fn }),
        this.getTasks(serviceName, { status: st, function: fn }),
      );
      );
      return this.$q.all(promisesTasks).then(flatten);
      return this.$q.all(promisesTasks).then(flatten);
    }
    }
    /**
    /**
     * Get task
     * Get task
     * @param {string} serviceName
     * @param {string} serviceName
     * @param {string} id
     * @param {string} id
     */
     */
    getTask(serviceName, id) {
    getTask(serviceName, id) {
      return this.OvhHttp.get(`/domain/${serviceName}/task/${id}`, {
      return this.OvhHttp.get(`/domain/${serviceName}/task/${id}`, {
        rootPath: 'apiv6',
        rootPath: 'apiv6',
      }).then((originalTasksDetails) => {
      }).then((originalTasksDetails) => {
        const tasksDetails = clone(originalTasksDetails);
        const tasksDetails = clone(originalTasksDetails);
        tasksDetails.status =
        tasksDetails.status =
          isString(tasksDetails.status) && tasksDetails.status.toUpperCase();
          isString(tasksDetails.status) && tasksDetails.status.toUpperCase();
        return tasksDetails;
        return tasksDetails;
      });
      });
    }
    }

File: packages/manager/apps/web/client/app/components/service-status/service-status-action.html [hunk 15]
```diff
@@ -38,13 +38,21 @@
     </dl>
     <oui-action-menu data-compact data-placement="end">
         <oui-action-menu-item
-            data-ng-if="$ctrl.isExpired() || !$ctrl.isInAutoRenew()"
+            data-ng-if="$ctrl.serviceInfos.state ==='expired' && ($ctrl.isExpired() || !$ctrl.isInAutoRenew())"
```
Comment (mohammed-zahaf) [inline packages/manager/apps/web/client/app/components/service-status/service-status-action.html:41]: I guess we could pproced with some refactoring to make code more cleanup:
```suggestion
            data-ng-if="$ctrl.isServiceInfosExpired() && ($ctrl.isExpired() || !$ctrl.isInAutoRenew())"
            
            Into controller:
            isServiceInfosExpired() {
               this.serviceInfos.state === SERVICE_INFO_STATS.EXPIRED;
            }
            
            Into constant file:
            export default SERVICE_INFO_STATS = {
              EXPIRED : 'expired',
              RESTORABLE: 'restorable',
              ..........(to complete)
            }
```

**Review by mohammed-zahaf — APPROVED** [review]

**Comment by anooparveti** [comment]
Pulling trabslations

**Review by anooparveti — APPROVED** [review]

**Pull request merged**